lightning/src/ln/channelmanager.rs

```diff
@@ -6205,6 +6207,16 @@ where
 										payment_preimage
 									} else { fail_htlc!(claimable_htlc, payment_hash); }
 								} else { None };
+
+								// Authenticate the PaymentContext received over a BlindedPaymentPath
+								if let Some(payment_context) = payment_context.as_ref() {
+									if let Some((hmac, nonce)) = authentication {
+										if payment_context.verify_for_offer_payment(hmac, nonce, &self.inbound_payment_key).is_err() {
+											fail_htlc!(claimable_htlc, payment_hash);
```

Comment:
Hmmm... we could, but it would require having access to the `ExpandedKey`. And that would leak into the public `onion_payment::peel_payment_onion` function's interface, too, IIUC. Maybe that is ok?

Reply:
Yeah, I'm inclined to update `NodeSigner` as you mentioned. Also, instead of touching `peel_payment_onion`, we could do authenticate a little earlier when parsing the `InboundOnionPayload`. I guess this is more what you meant? It also has a `NodeSigner` and the entire `ReceiveTlvs`.

Reply:
Updated accordingly.